tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/TmailEmailContactSearchEngineMailet.java

```diff
@@ -0,0 +1,30 @@
+package com.linagora.tmail.james.jmap;
+
+import com.linagora.tmail.james.jmap.model.EmailAddressContact;
+import com.linagora.tmail.james.jmap.model.EmailAddressContactSearchEngine;
+import org.apache.james.jmap.api.model.AccountId;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMailet;
+import reactor.core.publisher.Flux;
+
+import javax.inject.Inject;
+import javax.mail.MessagingException;
+import java.util.UUID;
+
+public class TmailEmailContactSearchEngineMailet extends GenericMailet {
+    final EmailAddressContactSearchEngine emailAddressContactSearchEngine;
+
+    @Inject
+    public TmailEmailContactSearchEngineMailet(EmailAddressContactSearchEngine emailAddressContactSearchEngine) {
+        this.emailAddressContactSearchEngine = emailAddressContactSearchEngine;
+    }
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        Flux.fromIterable(mail.getRecipients())
+            .map(recipient -> emailAddressContactSearchEngine
+                 .index(AccountId.fromString(mail.getMaybeSender().asString()),
+                        new EmailAddressContact(UUID.fromString(recipient.getLocalPart()),
+                                                recipient.getLocalPart()+"@"+recipient.getDomain().asString())));
```

Comment:
Can we avoid string manipulation outside of the domain model?

This breaks the encapsulation brought in by object origented programming.